pkg/provider/gitlab/gitlab.go

```diff
@@ -226,6 +247,31 @@ func (g *Gitlab) GetSecretMap(ctx context.Context, ref esv1beta1.ExternalSecretD
 	return secretData, nil
 }
 
+func isValidSecret(checkName bool, ref esv1beta1.ExternalSecretFind, environment string, data *gitlab.ProjectVariable) (bool, string) {
+	if environment != "" && environment != "*" {
+		// as of now gitlab does not support filtering of EnvironmentScope through the api call
+		if data.EnvironmentScope != environment {
+			return false, ""
+		}
+	}
```

Comment:
does this method exists to filter out variables not matching the desired `environment_scope`. If so, I have 2 questsions:
- does this method support "all" for `environment_scope` from the point of gitlab ci server
- does this method support "no" `environment_scope` from the point of eso gitlab provider


Either way, the comment in line 252 is obsolete, isn't it ?

info: filtering by `environment_scope` for the gitlab provider is supported since ESO v0.0.6

Reply:
The comment is not obsolete. In contrast to the GetVariable, the ListVariables api in Gitlab does not support filtering for environment scope on the server side, thats why it is done here.
As for the questions: It should cover those cases.